### PR TITLE
[DEVOPS-1751] Pipeline for publishing java sdk to gradle

### PR DESCRIPTION
**Pull request by michalchecinski**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

- **.github/workflows/build-java.yml:** Run build on `rc` and `hotfix-rc` and on PR. Change Publish task to package.
- **.github/workflows/publish-java.yml** Add Publish workflow
- **.github/workflows/version-bump.yml** Add Java SDK to version bump workflow
- **languages/java/build.gradle** Change name of the package to `sdk-secrets`. Add `https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/` as repository to publish.

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

File: languages/java/build.gradle
Comment (mzieniukbw) [inline languages/java/build.gradle:1]: We should publish with java docs (and optionally source code), maybe add:
```
java {
    withJavadocJar()
    withSourcesJar()
}
```

File: languages/java/build.gradle [hunk 1]
```diff
@@ -19,14 +19,14 @@ repositories {
...
-    description = 'BitwardenSDK'
+    description = 'Bitwarden Secrets Manager Java SDK'
     java.sourceCompatibility = JavaVersion.VERSION_1_8
 
     publishing {
         publications {
             maven(MavenPublication) {
                 groupId = 'com.bitwarden'
-                artifactId = 'sdk'
+                artifactId = 'sdk-secrets'
```
Comment (mzieniukbw) [inline languages/java/build.gradle:29]: i though this sdk repo is not specific to secrets manager, but it's for everyone ?

**Review by mzieniukbw — CHANGES_REQUESTED** [review]
<del>I don't think the intention was to make this Java sdk to be Secrets Manager specific.</del>
See below for my other comments

**Review by mimartin12 — COMMENTED** [review]
Nothing from my side that @mzieniukbw hasn't already by brought up 👍

File: .github/workflows/build-java.yml [hunk 2]
```diff
@@ -60,10 +63,10 @@ jobs:
...
 
-      - name: Publish Maven
+      - name: Build Maven
         uses: gradle/actions/setup-gradle@ec92e829475ac0c2315ea8f9eced72db85bb337a # v3.0.0
         with:
-          arguments: publish
+          arguments: package
           build-root-directory: languages/java
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (michalchecinski) [reply .github/workflows/build-java.yml]: Right! Good catch.

File: .github/workflows/build-java.yml [hunk 3]
```diff
@@ -60,10 +63,10 @@ jobs:
           name: libbitwarden_c_files-x86_64-pc-windows-msvc
           path: languages/java/src/main/resources/win32-x86-64
 
-      - name: Publish Maven
+      - name: Build Maven
         uses: gradle/actions/setup-gradle@ec92e829475ac0c2315ea8f9eced72db85bb337a # v3.0.0
         with:
-          arguments: publish
+          arguments: package
```
[the remark this answers is not in the document]
Comment (michalchecinski) [reply .github/workflows/build-java.yml]: Fixed 😃

**Review by mzieniukbw — APPROVED** [review]
lgtm

**Review by mzieniukbw — DISMISSED** [review]

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**